src/backend/env/mod.rs

```diff
@@ -579,7 +579,9 @@ impl State {
 
     pub fn init(&mut self) {
         assets::load();
-        match token::balances_from_ledger(&mut self.memory.ledger.iter().map(|(_, tx)| tx)) {
+        match token::balances_from_ledger(
```

Comment:
If we go with the plan I proposed, we would need to embed the JSON file with balances and check that its hash matches the "decentralized consensus" value. Then load the balances from that JSON.

Reply:
Ah so you want to do it all in the same release? But how can we prevent balance changes between the upgrade and proposal with a json file?